mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMessageIdDAO.java

```diff
@@ -78,14 +72,13 @@ private PreparedStatement prepareSelect(Session session) {
             .where(eq(ATTACHMENT_ID_AS_UUID, bindMarker(ATTACHMENT_ID_AS_UUID))));
     }
 
-    public CompletableFuture<Collection<MessageId>> getOwnerMessageIds(AttachmentId attachmentId) {
+    public Flux<MessageId> getOwnerMessageIds(AttachmentId attachmentId) {
         Preconditions.checkArgument(attachmentId != null);
-        return cassandraAsyncExecutor.execute(
-            selectStatement.bind()
-                .setUUID(ATTACHMENT_ID_AS_UUID, attachmentId.asUUID()))
-            .thenApply(resultSet -> cassandraUtils.convertToStream(resultSet)
-                .map(this::rowToMessageId)
-                .collect(Guavate.toImmutableSet()));
+        return cassandraAsyncExecutor.executeReactor(
+                selectStatement.bind()
+                    .setUUID(ATTACHMENT_ID_AS_UUID, attachmentId.asUUID()))
+            .flatMapMany(Flux::fromIterable)
+            .map(this::rowToMessageId);
```

Comment:
when you loose the toImmutableSet conversion, you loose the unicity in the collection. Do you meant it?

Reply:
No. I didn't notice it because of the method return time. Either it's useful and we lack a test or it's useless, I don't know.

Reply:
it's seems at least useful as a very simple optimization. Do you think you can add a test and restore this behavior?

Reply:
this commit is still not fixed?

Reply:
I have added a test case, but there is no regression here